File: app/src/main/java/com/sd/www/cache/MMKVCacheStore.java
package com.sd.www.cache;

import android.content.Context;

import androidx.annotation.NonNull;

import com.sd.lib.cache.Cache;
import com.sd.lib.cache.CacheInfo;
import com.tencent.mmkv.MMKV;

/**
 * 自定义CacheStore
 */
public class MMKVCacheStore implements Cache.CacheStore {
    private final MMKV _mmkv;

    public MMKVCacheStore(Context context) {
        MMKV.initialize(context);
        _mmkv = MMKV.defaultMMKV();
    }

    @Override
    public boolean putCache(@NonNull String key, @NonNull byte[] value, @NonNull CacheInfo info) {
        return _mmkv.encode(key, value);
    }

    @Override
    public byte[] getCache(@NonNull String key, @NonNull CacheInfo info) {
        return _mmkv.decodeBytes(key);
    }

    @Override
    public boolean removeCache(@NonNull String key, @NonNull CacheInfo info) {
        _mmkv.remove(key);
        return true;
    }

    @Override
    public boolean containsCache(@NonNull String key, @NonNull CacheInfo info) {
        return _mmkv.contains(key);
    }
}

File: app/src/main/java/com/sd/demo/cache/MMKVCacheStore.kt
package com.sd.demo.cache

import android.content.Context
import com.sd.lib.cache.Cache.CacheStore
import com.tencent.mmkv.MMKV

/**
 * 自定义CacheStore
 */
class MMKVCacheStore(context: Context) : CacheStore {
    private val _mmkv: MMKV

    init {
        MMKV.initialize(context)
        _mmkv = MMKV.defaultMMKV()
    }

    override fun putCache(key: String, value: ByteArray): Boolean {
        return _mmkv.encode(key, value)
    }

    override fun getCache(key: String): ByteArray? {
        return _mmkv.decodeBytes(key)
    }

    override fun removeCache(key: String) {
        _mmkv.remove(key)
    }

    override fun containsCache(key: String): Boolean {
        return _mmkv.contains(key)
    }
}